be/src/exec/es/es_scan_reader.cpp

```diff
@@ -79,18 +81,30 @@ ESScanReader::ESScanReader(const std::string& target, const std::map<std::string
         _exactly_once = true;
         std::stringstream scratch;
         // just send a normal search  against the elasticsearch with additional terminate_after param to achieve terminate early effect when limit take effect
-        scratch << _target << REQUEST_SEPARATOR << _index << REQUEST_SEPARATOR << _type << "/_search?"
-                << "terminate_after=" << props.at(KEY_TERMINATE_AFTER) << REQUEST_PREFERENCE_PREFIX << _shards << "&"
-                << filter_path;
+        if (_type.empty()) {
```

Comment:
how about define a static patten like 
static const char* pattern = "{}/{}/_search?terminate_after={}&preference=_shards:{}&{}"
then you can use fmt::format(pattern, _target, _index, props.at(), _shards)
I think it is easy of read